### PR TITLE
SAK-42786: Site Archive Tool -> Error archiving when assignment.merge.import.submissions=true

### PR DESCRIPTION
**Pull request by jesusmmp**

### CODE REVIEW

File: assignment/impl/src/java/org/sakaiproject/assignment/impl/AssignmentServiceImpl.java [hunk 1]
```diff
@@ -820,15 +820,18 @@ private Assignment mergeAssignment(final String siteId, final Element element, f
...
-                // here it's imported exactly as it was including all submissions
-                // except for group submissions as group ids will never be the same
+            if (serverConfigurationService.getBoolean(SAK_PROP_ASSIGNMENT_IMPORT_SUBMISSIONS, false)) {
                 Set<AssignmentSubmission> submissions = assignmentFromXml.getSubmissions();
-                List<String> submitters = submissions.stream().flatMap(s -> s.getSubmitters().stream()).map(AssignmentSubmissionSubmitter::getSubmitter).collect(Collectors.toList());
-                // only if all submitters can be found do we import submissions
-                if (submitters.containsAll(userDirectoryService.getUsers(submitters))) {
-                    submissions.forEach(s -> s.setId(null));
-                    submissions.forEach(s -> s.getSubmitters().forEach(u -> u.setId(null)));
+                if (submissions != null) {
```
Comment (ern) [inline assignment/impl/src/java/org/sakaiproject/assignment/impl/AssignmentServiceImpl.java]: submissions should never be null here, why a null check isn't needed.

Comment (ern) [reply assignment/impl/src/java/org/sakaiproject/assignment/impl/AssignmentServiceImpl.java]: If the submissions are coming back null as a result of deserialization then I would try adding:
`@JsonSetter(nulls = Nulls.AS_EMPTY)`

Comment (jesusmmp) [reply assignment/impl/src/java/org/sakaiproject/assignment/impl/AssignmentServiceImpl.java]: > submissions should never be null here, why a null check isn't needed.

If you archive a site with an assignment with no submissions, mergeAssignment methods fails.

**Comment by jesusmmp** [comment]
> If the group id's are never the same, does it make sense to import submissions for groups that don't exist?

Groups don't exist if you import in another site, but groups exist if you import in the same site.
We need a way to archive all information about the assignment (thinking the possibility an assignment is deleted and we want to recover all information).
How can we do this?

File: assignment/impl/src/java/org/sakaiproject/assignment/impl/AssignmentServiceImpl.java (hunk 1 above)
Comment (jesusmmp) [reply assignment/impl/src/java/org/sakaiproject/assignment/impl/AssignmentServiceImpl.java]: Hi @ern, is this ok?

**Comment by mpellicer** [comment]
@ern Any thoughts on this Jira? 

The PR looks good to me

**Pull request merged**